prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5583,17 +5602,22 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         if !self.init_info.is_vir_place_accessible(&dst, location) {
             let mut alloc_stmts = self.encode_havoc(&dst);
             let dst_field = dst.clone().field(field.clone());
-            let acc = vir::Expr::acc_permission(dst_field, vir::PermAmount::Write);
+            let acc = vir::Expr::acc_permission(dst_field.clone(), vir::PermAmount::Write);
             alloc_stmts.push(vir::Stmt::Inhale(acc));
             match vir_assign_kind {
                 vir::AssignKind::Copy => {
                     if field.typ.is_ref() {
-                        // TODO: Inhale the predicate rooted at dst_field
-                        return Err(SpannedEncodingError::unsupported(
-                            "the encoding of this reference copy has not \
-                            been implemented",
-                            self.mir_encoder.get_span_of_location(location),
-                        ));
+                        if field.typed_ref_name() == Some("str".to_string()) {
```

Comment:
Any way to access the MIR type here? If not, perhaps this equality check should be something implemented in `vir::Type`, similar to `is_ref`?

Reply:
I refactored this to instead just pass a flag to this enabling reference copies.

The flag right now is true for references to literals, as per Zulipchat https://prusti.zulipchat.com/#narrow/stream/288456-prusti-developers/topic/Supporting.20String.20Literals/near/250032719